File: MATTHEW/leetcode/AIcoding/2.Gemini-2.0/python3_solutions/1941-check-if-all-characters-have-equal-number-of-occurrences.py
class Solution:
    def areOccurrencesEqual(self, s: str) -> bool:
        counts = {}
        for char in s:
            counts[char] = counts.get(char, 0) + 1

        first_count = list(counts.values())[0]
        for count in counts.values():
            if count != first_count:
                return False

        return True

File: leetcode/AIcoding/2.Gemini-2.0/java_solutions/1941-check-if-all-characters-have-equal-number-of-occurrences.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public boolean areOccurrencesEqual(String s) {
        Map<Character, Integer> counts = new HashMap<>();
        for (char c : s.toCharArray()) {
            counts.put(c, counts.getOrDefault(c, 0) + 1);
        }

        int firstCount = -1;
        for (int count : counts.values()) {
            if (firstCount == -1) {
                firstCount = count;
            } else if (count != firstCount) {
                return false;
            }
        }

        return true;
    }
}